tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,521 @@
+            AppActions.setSidebarLoaded(true);
+            return waitForPromisesToResolve();
+        })
+        .then(() => renderedApp);
+}
+
+describe('Unread Indicators', () => {
+    afterEach(Onyx.clear);
+
+    it('Display bold in the LHN for unread chat and new line indicator above the chat message when we navigate to it', () => {
+        let renderedApp;
+        return signInAndGetAppWithUnreadChat()
+            .then((testInstance) => {
+                renderedApp = testInstance;
+
+                // Verify no notifications are created for these older messages
+                expect(LocalNotification.showCommentNotification.mock.calls.length).toBe(0);
+
+                // Verify the sidebar links are rendered
+                const sidebarLinks = renderedApp.queryAllByA11yLabel('List of chats');
+                expect(sidebarLinks.length).toBe(1);
+                expect(isDrawerOpen(renderedApp)).toBe(true);
+
+                // Verify there is only one option in the sidebar
+                const optionRows = renderedApp.getAllByA11yHint('Navigates to a chat');
+                expect(optionRows.length).toBe(1);
+
+                // And that the text is bold
+                const displayNameText = renderedApp.getByA11yLabel('Chat user display names');
+                expect(lodashGet(displayNameText, ['props', 'style', 0, 'fontWeight'])).toBe(fontWeightBold);
+
+                return navigateToSidebarOption(renderedApp, 0);
+            })
+            .then(() => {
+                // Verify that the report screen is rendered and the drawer is closed
+                expect(isDrawerOpen(renderedApp)).toBe(false);
+
+                // That the report actions are visible along with the created action
+                const createdAction = renderedApp.getByA11yLabel('Chat welcome message');
+                expect(createdAction).toBeTruthy();
+                const reportComments = renderedApp.getAllByA11yLabel('Chat message');
+                expect(reportComments.length).toBe(9);
+
+                // Since the last read sequenceNumber is 1 we should have an unread indicator above the next "unread" action which will
+                // have a sequenceNumber of 2
+                const unreadIndicator = renderedApp.queryAllByA11yLabel('New message line indicator');
+                expect(unreadIndicator.length).toBe(1);
+                const sequenceNumber = lodashGet(unreadIndicator, [0, 'props', 'data-sequence-number']);
+                expect(sequenceNumber).toBe(2);
+
+                // Scroll up and verify that the "New messages" badge appears
+                scrollUpToRevealNewMessagesBadge(renderedApp);
+                expect(isNewMessagesBadgeVisible(renderedApp)).toBe(true);
+            });
+    });
+
+    it('Clear the new line indicator and bold when we navigate away from a chat that is now read', () => {
```

Comment:
I usually try to write these so that can be read from the CLI easier like a sentence combining both the group description and the test description. So, for this, the group description is "Unread Indicators". If you then add the test description to that, you get "Unread Indicators Clear the new line indicator and bold when we navigate away from a chat that is now read" which reads very weird. 

My suggested edit would be:

```suggestion
    it('are removed when a chat is read and navigated away from', () => {
```

That way, you end up with "Unread Indicators are removed when a chat is read and navigated away from" which is very readable.

This is just a NAB, and something to consider.

Reply:
Makes sense. I guess some of these are testing a few different things so there's not always a single subject.

Reply:
In my LHN tests, I even used nested `describe()` to help with that. Maybe there is a good way to do this in your tests.

![image](https://user-images.githubusercontent.com/1228807/190826159-f99dde22-239a-4d78-ad2b-d359af36b87c.png)

It's easy to see how tests are grouped together and why (different description for each subject kinda)